### ISSUE TITLE
Unable to save trained model.

### ISSUE THREAD

**Issue by leowang7** [idealo/image-super-resolution#102]
Hi,
I was successfully ran the training code, but at the end it seems that the model cannot be saved. After I did some research in Google, I found the issue probably related to tensorflow's bug, which is 'failed to serialize as JSON'. I got this warning too, please see below.

Did anyone encounter the same issue, any workaround? Thanks a lot. 

----
**WARNING:tensorflow:Model failed to serialize as JSON. Ignoring... can't pickle _thread.RLock objects**
Epoch 0/1
Current learning rate: 0.00039999998989515007
100%|██████████| 1/1 [00:12<00:00, 12.36s/it]
Epoch 0 took       12.4s
160/1 [==========================================================================] - 11s 71ms/sample - loss: 0.1855 - generator_loss: 0.1919 - discriminator_loss: 0.6483 - feature_extractor_loss: 0.9845 - feature_extractor_1_loss: 6.6397 - generator_PSNR_Y: 14.3487
val_PSNR_Y is NOT among the model metrics, removing it.
{'val_loss': 0.3240301303565502, 'val_generator_loss': 0.19187796, 'val_discriminator_loss': 0.64834464, 'val_feature_extractor_loss': 0.9844631, 'val_feature_extractor_1_loss': 6.6397047, 'val_generator_PSNR_Y': 14.348654, 'train_d_real_loss': 0.9044446, 'train_d_real_accuracy': 0.240625, 'train_d_fake_loss': 0.91425353, 'train_d_fake_accuracy': 0.70671874, 'train_loss': 0.14742404, 'train_generator_loss': 0.076132715, 'train_discriminator_loss': 0.6488205, 'train_feature_extractor_loss': 0.44856423, 'train_feature_extractor_1_loss': 2.9352493, 'train_generator_PSNR_Y': 16.832394}

**Comment by kennethchanchifung** [comment]
same here, did you solve the problem?

**Comment by leowang7** [comment]
> same here, did you solve the problem?

I was able to save trained model by using default Div2k data-set although I still got same warning.

**Comment by kennethchanchifung** [comment]
Finally I use docker and mount a host directory to solve the problem.

**Comment by gkrislara** [comment]
@kennethchanchifung  can you elaborate what you did?

**Comment by kennethchanchifung** [comment]
@gkrislara 
1. create docker image by the file Dockerfile.gpu, let say define the name as isr_image (you can define any name you want) 
2. Execute docker container by the following command
sudo docker run --gpus all -it -v {path of project of host}:{path of project of docker container} {docker image name} -c config.yml

Example
sudo docker run --gpus all -it -v /home/image-super-resolution/:/home/isr/ isr_image -c config.yml

**Comment by Timliuzhaolu** [comment]
Hi,

I also get the same warning. But I thought the model is saved in the weights folder. If I understand it correctly, that hd5y file on its own is capable of making predictions right? Can we just ignore that warning? Thanks!
